katcp/test/test_server.py

```diff
@@ -1014,13 +1014,59 @@ def test_sampling(self):
         self.server.ioloop.add_callback(self.server.clear_strategies, client_conn)
         self.server.sync_with_ioloop()
         self.client.assert_request_succeeds("sensor-sampling", "an.int",
-                                            args_equal=[b"an.int", b"none"])
-
+                                            args_equal=["an.int", "none"])
         # Check that we did not accidentally clobber the strategy datastructure
         # in the proccess
         self.client.assert_request_succeeds(
             "sensor-sampling", "an.int", "period", 0.125)
 
+    def test_assert_request_succeeds(self):
+        """Test exercises assert_request_succeeds it is a copy of test_sampling."""
```

Comment:
This tests seems overly complicated for testing `assert_request_succeeds`.  Could we simplify so that it is easier for the reader to see the required behaviour of `assert_request_succeeds`?
E.g., just using a standard request like `?sensor-value`, and removing all the sampling stuff.

Reply:
Agreed. This was just the quickest for me as I was adding the test cases to the `test_sampling`. 
It seems that we have a similar set of issues with `assert_request_fails` see https://github.com/ska-sa/katcp-python/blob/0133e9ee0f5ba0cf7d5731105ad83592eacb178a/katcp/testutils.py#L829 
Should I follow the approach of this PR for `assert_request_fails`

Reply:
It could be added to this PR, but you can decide.